src/amqp.lisp

```diff
@@ -845,3 +847,428 @@ retrieved has been processed"
+            (method-frame-basic-return/channel obj)
+            (method-frame-basic-return/id obj)
+            (method-frame-basic-return/reply-code obj)
+            (method-frame-basic-return/reply-text obj)
+            (method-frame-basic-return/exchange obj)
+            (method-frame-basic-return/routing-key obj))))
+
+(defclass method-frame-basic-deliver (method-frame)
+  ((channel :reader method-frame-basic-deliver/channel)
+   (id :initform #.+amqp-basic-deliver-method+
+       :reader method-frame-basic-deliver/id
+       :allocation :class
+       :documentation "Basic deliver method ID.")
+   (consumer-tag :type string
+                 :initarg :consumer-tag
+                 :reader method-frame-basic-deliver/consumer-tag
+                 :documentation "Identifier for the consumer, valid within the current channel.
+The server-assigned and channel-specific delivery tag")
+   (delivery-tag :type integer
+                 :initarg :delivery-tag
+                 :reader method-frame-basic-deliver/delivery-tag
+                 :documentation "Sequence number of confirmed message.")
+   (redeliveredp :type t
+                 :initarg :redeliveredp
+                 :reader method-frame-basic-deliver/redeliveredp
+                 :documentation "t when message is redelivered, after it was not successfully
+acknowledged by a consumer.")
+   (exchange :type string
+             :initarg :exchange
+             :reader method-frame-basic-deliver/exchange
+             :documentation "Exchange name.")
+   (routing-key :type string
+                :initarg :routing-key
+                :reader method-frame-basic-deliver/routing-key
+                :documentation "Message routing key."))
+  (:documentation "Basic deliver method frame."))
+
+(defun make-method-frame-basic-deliver (channel
+                                        consumer-tag
+                                        delivery-tag
+                                        redeliveredp
+                                        exchange
+                                        routing-key)
+  "Returns fresh object of METHOD-FRAME-BASIC-DELIVER.
+
+Arguments:
+
+CHANNEL => INTEGER, Rabbitmq channel.
+CONSUMER-TAG => STRING, identifier for the consumer, valid within the current channel.
+DELIVERY-TAG => INTEGER, sequence number of confirmed message.
+REDELIVEREDP => BOOLEAN, t when message is redelivered after it was not successfully acknowledged
+by a consumer.
+EXCHANGE => STRING, Exchange name.
+ROUTING-KEY => STRING, Message routing key."
+  (make-instance 'method-frame-basic-deliver
+                 :channel channel
+                 :consumer-tag consumer-tag
+                 :delivery-tag delivery-tag
+                 :redeliveredp redeliveredp
+                 :exchange exchange
+                 :routing-key routing-key))
+
+(defmethod print-object ((obj method-frame-basic-deliver) stream)
+  (print-unreadable-object (obj stream :type t :identity nil)
```

Comment:
If any of the slots are unbound when the object is printed, you might end up with an error while printing an object. This is particularly annoying when the debugger is printing it. There is a function in `misc.lisp` which handles this, please take a look at `PRINT-UNREADABLE-SAFELY`.

Reply:
Updated